pyasic/miners/base.py

```diff
@@ -560,5 +560,18 @@ def __init__(self, ip: str) -> None:
         if self._ssh_cls is not None:
             self.ssh = self._ssh_cls(ip)
 
+    async def upgrade_firmware(self, *, file: str = None, url: str = None, version: str = "latest", keep_settings: bool = True) -> bool:
```

Comment:
Version here can default to None, and just define it for any miner type that takes version like this.